### PR TITLE
Feature: Keytool

### PR DESCRIPTION
**Pull request by zlanyi**
- feature: generate Keytool files
- fix: profile.erb due to "RDECK_SSL_OPTS"

### CODE REVIEW

**Comment by zlanyi** [comment]
Hi,
i did a new pull request because of the previous version of pull request (260) got a very strange commit .

Cheers,
Zoltan

**Comment by zlanyi** [comment]
I hope all things okay.

**Comment by zlanyi** [comment]
Hi Guys,
any news here?

Cheers,
Zoltan

**Comment by jyaworski** [comment]
@zlanyi can you rebase on the current master? It looks good.

**Comment by zlanyi** [comment]
"spec/acceptance/rundeck_spec.rb:5:5: C: RSpec/MultipleExpectations: Too many expectations."

**Comment by alexjfisher** [comment]
@zlanyi You could probably disable that cop for now by adding
```
# rubocop:disable RSpec/MultipleExpectations
```
to the top of the file.  eg like in https://github.com/voxpupuli/puppet-archive/commit/fcfc08607376b774860cc8a0d8035a3c770e336e

Other than than, could you *rebase* your PR instead of merging from upstream?  If you're not that comfortable with git head on over to the #voxpupuli channel on Freenode IRC.  Plenty of people willing to help you out there.

Thanks

**Comment by zlanyi** [comment]
@jyaworski : i did so the rebase
git checkout master
git pull upstream master
git checkout keytools2
git rebase master

its wrong?

**Comment by alexjfisher** [comment]
Ideally, the end result would be one or two logical commits, not the 13 here https://github.com/voxpupuli/puppet-rundeck/pull/261/commits

`copy branch to keytool2 because of a strange commit from igalic` is my favorite! ;)

Perhaps try a `rebase -i master` and do some squashing.  But maybe best to make sure you have a safe copy of your branch first.  `git checkout keytools2; git checkout -b keytools2_backup; git checkout keytools2`

File: .rubocop.yml [hunk 1]
```diff
@@ -504,3 +504,7 @@ RSpec/DescribeClass:
 # Example length is not necessarily an indicator of code quality
 RSpec/ExampleLength:
   Enabled: False
+
```
Comment (alexjfisher) [inline .rubocop.yml]: The `.rubocop.yml` file is managed by modulesync and shouldn't be edited manually.  Could you disable the cop in the same manner as in https://github.com/voxpupuli/puppet-archive/commit/fcfc08607376b774860cc8a0d8035a3c770e336e by inserting the rubocop control comment at the first line of `spec/acceptance/rundeck_spec.rb`?

Comment (zlanyi) [inline .rubocop.yml]: done

**Comment by alexjfisher** [comment]
@zlanyi Great!  One last thing... You're now back up to 5 commits.  Could you squash them down to just a single one with a suitable commit message?

Thanks.

**Comment by zlanyi** [comment]
@alexjfisher : thanks your support, done

**Comment by alexjfisher** [comment]
@zlanyi I'm really, really sorry for not merging this yet... But... could you fix up the English in the README?
(You can use `git commit --amend` to avoid creating a new commit)
Thanks.

File: README.md [hunk 2]
```diff
@@ -65,6 +65,36 @@ Web context path to use, such as "/rundeck". http://host.domain:port/server_web_
 #####`ssl_enabled`
 Enable ssl for the Rundeck web application.
 
+#####`ssl_keyfile`
+#####`ssl_certfile`
+If ssl enabled you need set up this. You need provide the crt and key files in your profile oder role manifest.
```
Comment (bbriggs) [inline README.md]: This sentence needs a little more work in order to be clear to users. 

Let's start with something like this:

"If `ssl_enabled` is True, you must supply this parameter. It is recommended that you provide the `.crt` and `.key` files separately via other means, such as a role or profile manifest."

File: README.md [hunk 3]
```diff
@@ -65,6 +65,36 @@ Web context path to use, such as "/rundeck". http://host.domain:port/server_web_
...
+
+How to: eg: environments/role/manifests/rundeck.pp
```
Comment (bbriggs) [inline README.md]: Drop this line and include the sample file path it represents as a comment inside the appropriate code block.

File: README.md [hunk 4]
```diff
@@ -65,6 +65,36 @@ Web context path to use, such as "/rundeck". http://host.domain:port/server_web_
...
+
+class role::rundeck (
+...
```
Comment (bbriggs) [inline README.md]: This whole section is not displaying well :(

We should use code fences to markup and format code (the ``` at the start and end of the code block)

**Pull request merged**